src/agentic/actor_agents.py

```diff
@@ -1023,76 +1035,40 @@ def __init__(
         handle_turn_start: Callable[[Prompt, RunContext], None] = None,
         result_model: Type[BaseModel]|None = None,
         debug: DebugLevel = DebugLevel(os.environ.get("AGENTIC_DEBUG") or ""),
-        mock_settings: dict = None,  # Add mock_settings parameter
+        mock_settings: dict = None,
     ):
         self.name = name
         self.welcome = welcome or f"Hello, I am {name}."
         self.model = model or "gpt-4o-mini"
+        self.cancelled = False
+        
+        # Setup template path
         caller_frame = inspect.currentframe()
-        self.cancelled: bool = False
         if caller_frame:
             caller_file = inspect.getframeinfo(caller_frame.f_back).filename
             directory = os.path.dirname(caller_file)
-            # Get just the filename without extension
             base = os.path.splitext(os.path.basename(caller_file))[0]
-
-            # Create new path with .prompts.yaml extension
             template_path = os.path.join(directory, f"{base}.prompts.yaml")
-
-        # Get the file where Agent() was called
         self.template_path = template_path
+        
+        # Setup tools and other properties
         self._tools = []
-        if tools: #ack! If we take the default list shit goes bad!
+        if tools:
             self._tools.extend(tools)
-
+            
         self.max_tokens = max_tokens
         self.memories = memories
         self.debug = debug
         self._handle_turn_start = handle_turn_start
-        self.request_queues: dict[str, Queue] = {}
+        self.request_queues = {}
```

Comment:
This is how the proxy manages multiple requests. It creates an agent for each request ID, and separate thread Queue to receive events from that agent.